runhouse/servers/cluster_servlet.py

```diff
@@ -378,6 +390,13 @@ async def aperiodic_autostop_check(self):
             except Exception as e:
                 logger.error(f"Autostop check has failed: {e}")
 
+                # killing the cluster servlet only if log level is debug: this indicates that the cluster is created
```

Comment:
what if a user sets their RH_LOG_LEVEL env var to debug? do we still intend to kill the servlet then?

Reply:
This a good point, I guess it is not ideal, but I feel like it is the most elegant solution we've came up with. Hopefully we'll cover most of the failures on our own, since we set the log level to 'DEBUG' locally as well as in CI. This way we'll be in a pretty got place before release.

Indeed it is a bit risky to kill the servlet for users outside of our team. The benefit of doing this, however, is that users will probably reach to us with new errors that haven't encountered with. (The servlet errors are printed to the cluster logs regardless to the log level, but the chances the the user will notice and send them to us are pretty low, since the cluster is still up and working).

Another open question is how often will users set the log level to 'DEBUG', but this can be answered only once will have bunch of runhouse users. 

To sum this up - I think that for now its fine to proceed with this solution because:

1. We don't have much traffic / users at the moment.
2. This already found one bug that I fixed.
3. This will allow us to monitor other errors over time, especially now that we've introduced a lot of new changes and refactors. 

I think that we could bring it up for a further discussion at the beginning of the year. wdyt?

Reply:
sure we can go ahead with this for now, can you update the comment to just reflect that this is the case for debug mode and not necessary just for test/runhouse.

and let's also revisit this when @jlewitt1 is back too